src/arrayMethodJoin.js

```diff
@@ -4,8 +4,25 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    let stringSeparator = separator;
+    let result = '';
+
+    if (stringSeparator === null || typeof stringSeparator === 'object') {
+      stringSeparator = String(stringSeparator);
+    }
```

Comment:
Do you really need this?
Result will be the same with and without this check
<img width="206" alt="image" src="https://github.com/mate-academy/js_array-method-join/assets/86789161/8a566390-c0db-41e2-95f2-6ca0e87f40b8">